src/TokenHandler/AccountManager.mo

```diff
@@ -0,0 +1,408 @@
+          ignore depositRegistry.set(
+            p,
+            func(info) {
+              let amount = info.queued;
+              info.queued := 0;
+              queuedFunds -= amount;
+              info.underway := amount;
+              underwayFunds += amount;
+              true;
+            },
+          );
+          await* consolidate(p);
+          depositRegistry.unlock(p);
+        };
+      };
+    };
+
+    /// Processes the transfer of funds for withdrawal.
+    func processWithdrawTransfer(to : ICRC1.Account, amount : Nat) : async* {
+      #Ok : Nat;
+      #Err : ICRC1.TransferError or { #CallIcrc1LedgerError; #TooLowQuantity };
+    } {
+      if (amount <= fee_) return #Err(#TooLowQuantity);
+
+      try {
+        await icrc1Ledger.icrc1_transfer({
+          from_subaccount = null;
+          to = to;
+          amount = Int.abs(amount - fee_);
+          fee = ?fee_;
+          memo = null;
+          created_at_time = null;
+        });
+      } catch (err) {
+        #Err(#CallIcrc1LedgerError);
+      };
+    };
+
+    /// Initiates a withdrawal by transferring tokens to another account.
+    /// Returns ICRC1 transaction index and amount of transferred tokens (fee excluded).
+    public func withdraw(to : ICRC1.Account, amount : Nat) : async* Result.Result<(transactionIndex : Nat, withdrawnAmount : Nat), ICRC1.TransferError or { #CallIcrc1LedgerError; #TooLowQuantity }> {
+
+      totalWithdrawn_ += amount;
+
+      let callResult = await* processWithdrawTransfer(to, amount);
+
+      switch (callResult) {
+        case (#Ok txIdx) {
+          journal.push((Time.now(), ownPrincipal, #withdraw({ to = to; amount = amount })));
+          #ok(txIdx, amount - fee_);
+        };
+        case (#Err(#BadFee { expected_fee })) {
+          journal.push((Time.now(), ownPrincipal, #feeUpdated({ old = fee_; new = expected_fee })));
+          fee_ := expected_fee;
+          recalculateBacklog();
+          let retryResult = await* processWithdrawTransfer(to, amount);
+          switch (retryResult) {
+            case (#Ok txIdx) {
+              journal.push((Time.now(), ownPrincipal, #withdraw({ to = to; amount = amount })));
+              #ok(txIdx, amount - fee_);
+            };
+            case (#Err err) {
+              totalWithdrawn_ -= amount;
+              #err(err);
+            };
+          };
+        };
+        case (#Err err) {
+          totalWithdrawn_ -= amount;
+          #err(err);
+        };
+      };
+    };
+
+    func pushToBacklog(p : Principal, deposit : Nat) {
+      ignore depositRegistry.set(
+        p,
+        func(info) {
+          queuedFunds -= info.queued;
+          queuedFunds += deposit;
+          info.queued := deposit;
+          true;
+        },
+      );
+
+      backlog.push(p);
+    };
+
+    /// Recalculate the backlog after the fee change.
+    /// Reason: Some amounts in the backlog can be insufficient for consolidation.
+    func recalculateBacklog() {
```

Comment:
recalculateBacklog needs to know the old fee and the new fee.

Reply:
Yes, I will fix it